langchain/src/llms/bedrock.ts

```diff
@@ -0,0 +1,154 @@
+*/
+export interface BedrockInput {
+  /** Model to use.
+      For example, "amazon.titan-tg1-large", this is equivalent to the modelId property in the list-foundation-models api.
+  */
+  model: string;
+
+  /** The AWS region e.g. `us-west-2`.
+      Fallback to AWS_DEFAULT_REGION env variable or region specified in ~/.aws/config in case it is not provided here.
+  */
+  regionName?: string;
+
+  /** Temperature */
+  temperature?: number;
+
+  /** Max tokens */
+  maxTokens?: number;
+}
+
+export class Bedrock extends LLM implements BedrockInput {
+  model = "amazon.titan-tg1-large";
+
+  regionName?: string | undefined = undefined;
+
+  temperature?: number | undefined = undefined;
+
+  maxTokens?: number | undefined = undefined;
+
+  get lc_secrets(): { [key: string]: string } | undefined {
+    return {};
+  }
+
+  _llmType() {
+    return "bedrock";
+  }
+
+  constructor(fields?: Partial<BedrockInput> & BaseLLMParams) {
+    super(fields ?? {});
+
+    this.model = fields?.model ?? this.model;
+    const allowedModels = ["ai21", "anthropic", "amazon"];
+    if (!allowedModels.includes(this.model.split(".")[0])) {
+      throw new Error(
+        `Unknown model: '${this.model}', only these are supported: ${allowedModels}`
+      );
+    }
+    this.regionName =
+      fields?.regionName ?? getEnvironmentVariable("AWS_DEFAULT_REGION");
+    this.temperature = fields?.temperature ?? this.temperature;
+    this.maxTokens = fields?.maxTokens ?? this.maxTokens;
+  }
+
+  /** Call out to Bedrock service model.
+    Arguments:
+      prompt: The prompt to pass into the model.
+
+    Returns:
+      The string generated by the model.
+
+    Example:
+      response = model.call("Tell me a joke.")
+  */
+  async _call(prompt: string): Promise<string> {
+    const { createSignedFetcher } = await Bedrock.imports();
+
+    const signedFetcher = createSignedFetcher({
+      service: "bedrock",
+      region: this.regionName,
+    });
+
+    const url = `https://bedrock.${this.regionName}.amazonaws.com/model/${this.model}/invoke`;
+    const provider = this.model.split(".")[0];
+    const inputBody = LLMInputOutputAdapter.prepareInput(provider, prompt);
+
+    const response = await this.caller.call(
+      async () =>
+        await signedFetcher(url, {
+          method: "post",
+          body: JSON.stringify(inputBody),
+          headers: {
+            "Content-Type": "application/json",
+            accept: "application/json",
+          },
+        })
+    );
+
+    if (response.status < 200 || response.status >= 300) {
+      throw Error(
+        `Failed to access underlying url '${url}': got ${response.status} ${
+          response.statusText
+        }: ${await response.text()}`
+      );
+    }
+    const responseJson = await response.json();
+    const text = LLMInputOutputAdapter.prepareOutput(provider, responseJson);
+    return text;
+  }
+
+  /** @ignore */
+  static async imports(): Promise<{ createSignedFetcher: any }> {
```

Comment:
If you set up the entrypoints correctly (which it looks like you have), you can actually just put this at the top.

However, had a look at the package - it's not very popular and it doesn't look like it's doing much more than wrap the default AWS signature SDK. Could we just use that?

https://github.com/zirkelc/aws-sigv4-fetch/blob/main/src/create-signed-fetcher.ts#L14

Reply:
Yeah, that's what I was thinking - they're MIT licensed but I worry about their long-term stability and I would rather depend directly on AWS' package.

Reply:
That sounds good! I'm also fine with inlining and refactoring later.

Not sure what you mean by hard dependency though - if separated out correctly into an entrypoint one can import the rest of LangChain without importing this file and avoid relying on @aws-sdk/signature-v4. 

I can resolve any build issues at the end though, so don't worry about that too much.